internal/content/content.go

```diff
@@ -0,0 +1,382 @@
+				shouldDelete, err := console.PromptYesNo(ctx, msg, false)
+				if err != nil {
+					return err
+				}
+				if shouldDelete {
+					for _, file := range localOnlyFiles {
+						filePath := filepath.Join(config.Directory, file)
+						if err := fsys.Remove(filePath); err != nil {
+							fmt.Fprintf(os.Stderr, "failed to delete %s: %v\n", filePath, err)
+						} else {
+							fmt.Printf("Deleted local %s %s\n", config.DisplayName, strings.TrimSuffix(file, ".sql"))
+						}
+					}
+				} else {
+					fmt.Printf("Skipped deletion of %d local %ss\n", len(localOnlyFiles), config.DisplayName)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func DownloadOne(ctx context.Context, fsys afero.Fs, itemId string, contentType ContentType) error {
+	if err := flags.LoadConfig(fsys); err != nil {
+		return err
+	}
+
+	id, err := uuid.Parse(itemId)
+	if err != nil {
+		return fmt.Errorf("invalid %s ID: %w", GetConfig(contentType).DisplayName, err)
+	}
+	resp, err := utils.GetSupabase().V1ProjectGetSnippetWithResponse(ctx, flags.ProjectRef, id)
+	if err != nil {
+		return errors.Errorf("failed to download %s: %w", GetConfig(contentType).DisplayName, err)
+	}
+	if resp.JSON200 == nil {
+		return errors.New("unexpected error downloading SQL " + GetConfig(contentType).DisplayName + ": " + string(resp.Body))
+	}
+
+	if !isCorrectResponseType(resp.JSON200.Type, contentType) || resp.JSON200.Visibility != api.SnippetResponseVisibilityProject {
+		return errors.New("requested item is not a project SQL " + GetConfig(contentType).DisplayName)
+	}
+
+	if resp.JSON200.Content.Sql != nil {
+		fmt.Println(*resp.JSON200.Content.Sql)
+	}
+	return nil
+}
+
+func listRemote(ctx context.Context, contentType ContentType) (map[string]uuid.UUID, error) {
+	listType := getListParamsType(contentType)
+	opts := api.V1ProjectListSnippetsParams{Type: &listType}
+	resp, err := utils.GetSupabase().V1ProjectListSnippetsWithResponse(ctx, flags.ProjectRef, &opts)
+	if err != nil {
+		return nil, err
+	}
+	if resp.JSON200 == nil {
+		return nil, errors.New(string(resp.Body))
+	}
+	m := map[string]uuid.UUID{}
+	for _, s := range resp.JSON200.Data {
+		if !isCorrectType(s.Type, contentType) || s.Visibility != api.SnippetListDataVisibilityProject {
+			continue
+		}
+		id, err := uuid.Parse(s.Id)
+		if err == nil {
+			m[s.Name] = id
+		}
+	}
+	return m, nil
+}
+
+func upsert(ctx context.Context, contentType ContentType, id uuid.UUID, name, sql string) error {
+	content := map[string]interface{}{
+		"favorite":       false,
+		"schema_version": schemaVersion,
+		"sql":            sql,
+	}
+	body := api.UpsertContentBody{
+		Id:         id.String(),
+		Name:       name,
+		OwnerId:    0,
+		Type:       getUpsertBodyType(contentType),
+		Visibility: api.UpsertContentBodyVisibilityProject,
+		Content:    &content,
+	}
+	resp, err := utils.GetSupabase().V1ProjectUpsertSnippetWithResponse(ctx, flags.ProjectRef, id, body)
```

Comment:
upsert really means update or insert. The operation id for this openapi spec should be `v1-project-update-snippet` for clarity